x/concentrated-liquidity/math/math_test.go

```diff
@@ -323,6 +337,39 @@ func (suite *ConcentratedMathTestSuite) TestGetLiquidityFromAmounts() {
 			amount1Desired:    sdk.NewInt(5000000000),
 			expectedLiquidity: "1517882343.751510418088349649",
 		},
+		"full range, price proportional to amounts, equal liquidities (some rounding error) price of 4": {
+			currentSqrtP:   sqrt(sdk.NewDec(4)),
+			sqrtPHigh:      cltypes.MaxSqrtPrice,
+			sqrtPLow:       cltypes.MinSqrtPrice,
+			amount0Desired: sdk.NewInt(4),
+			amount1Desired: sdk.NewInt(16),
+
+			expectedLiquidity:  sdk.MustNewDecFromStr("8.000000000000000001").String(),
+			expectedLiquidity0: sdk.MustNewDecFromStr("8.000000000000000001"),
+			expectedLiquidity1: sdk.MustNewDecFromStr("8.000000004000000002"),
+		},
+		"full range, price proportional to amounts, equal liquidities (some rounding error) price of 2": {
+			currentSqrtP:   sqrt(sdk.NewDec(2)),
+			sqrtPHigh:      cltypes.MaxSqrtPrice,
+			sqrtPLow:       cltypes.MinSqrtPrice,
+			amount0Desired: sdk.NewInt(1),
+			amount1Desired: sdk.NewInt(2),
+
+			expectedLiquidity:  sdk.MustNewDecFromStr("1.414213562373095049").String(),
+			expectedLiquidity0: sdk.MustNewDecFromStr("1.414213562373095049"),
+			expectedLiquidity1: sdk.MustNewDecFromStr("1.414213563373095049"),
+		},
+		"not full range, price proportional to amounts, non equal liquidities": {
+			currentSqrtP:   sqrt(sdk.NewDec(2)),
+			sqrtPHigh:      sqrt(sdk.NewDec(3)),
+			sqrtPLow:       sqrt(sdk.NewDec(1)),
+			amount0Desired: sdk.NewInt(1),
+			amount1Desired: sdk.NewInt(2),
+
+			expectedLiquidity:  sdk.MustNewDecFromStr("4.828427124746190095").String(),
+			expectedLiquidity0: sdk.MustNewDecFromStr("7.706742302257039729"),
+			expectedLiquidity1: sdk.MustNewDecFromStr("4.828427124746190095"),
+		},
```

Comment:
I really appreciate these added test cases to help better demonstrate this!

Do we know where the error in the 10^-9 place is coming from? I assume its because we divide in the liquidity1 calculation right?

Reply:
It actually seems to be correct:

```
>>> 16 / (Decimal("4").sqrt() - Decimal("0.000000000000000001").sqrt())
Decimal('8.000000004000000002000000001')
```

It might be the closeness to 0 vs infinity